src/file-upload/index.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { applyDisplayName } from '../internal/utils/apply-display-name';
+import { FileUploadProps } from './interfaces';
+import InternalFileUpload from './internal';
+import useBaseComponent from '../internal/hooks/use-base-component';
+import { getExternalProps } from '../internal/utils/external-props';
+
+export { FileUploadProps };
+
+const FileUpload = React.forwardRef(
+  (
+    {
+      multiple = false,
+      showFileSize = false,
+      showFileLastModified = false,
+      showFileThumbnail = false,
```

Comment:
The current `InternalFileUploadProps` type definition if very far from my understanding of "explicit"

```js
type InternalFileUploadProps = SomeRequired<
  FileUploadProps,
  'multiple' | 'showFileSize' | 'showFileLastModified' | 'showFileThumbnail'
> &
  InternalBaseComponentProps;
```

This could be just  `type InternalFileUploadProps = FileUploadProps & InternalBaseComponentProps`

But I will leave it up to you. Anyone can change it later when they stumble upon this

Reply:
Alright, I will change it now.